stdlib/list.ml

```diff
@@ -197,16 +197,26 @@ let partition p l =
   | x :: l -> if p x then part (x :: yes) no l else part yes (x :: no) l in
   part [] [] l
 
-let rec split = function
-    [] -> ([], [])
-  | (x,y)::l ->
-      let (rx, ry) = split l in (x::rx, y::ry)
+let rev_split xys =
+  let rec _rev_split xs ys = function
+    | [] -> xs,ys
```

Comment:
Minor nitpick: the usual style (in the core distrib at least) is to have a whitespace after the comma.